Source/WebCore/html/HTMLSlotElement.cpp

```diff
@@ -145,6 +145,13 @@ static void flattenAssignedNodes(Vector<Ref<Node>>& nodes, const HTMLSlotElement
     }
 }
 
+bool HTMLSlotElement::hasFlattenedSlottedContent() const
+{
+    Vector<Ref<Node>> nodes;
+    flattenAssignedNodes(nodes, *this);
+    return !nodes.isEmpty();
```

Comment:
Seems inefficient to build the list of nodes just to check if it’s empty. Can we do it more efficiently?